### DISCUSSION TITLE
Visiting Nuxtjs website automatically logs into Directus account

### DISCUSSION THREAD

**Discussion by julianftw** [directus/directus#15689] [Archive]
### Describe the Bug

Hi,

I have a questions regarding this tutorial from your website: https://directus.io/guides/get-started-building-a-nuxt-2-website-with-directus/

I followed the steps and everything is working fine, however there is a behavior that is not quite clear to me if it is intended or a bug. 

When I run Directus and Nuxtjs and then visit they Nuxtjs page (http://localhost:3000), I will be logged into the user on the Directus page (http://localhost:8055) with the credentials used in the nuxt.config.js even if I was previously logged into another account. 

I made some changes to the original repository and set up different users, but even when I clone the examples and don't change anything it's still the same.

The part from nuxt.config.js for Directus:

`directus: {
    url: process.env.DIRECTUS_URL,
    auth: {
      email: process.env.DIRECTUS_EMAIL,
      password: process.env.DIRECTUS_PASSWORD,
      // token: process.env.DIRECTUS_STATIC_TOKEN,
    },
  },`

The following message is displayed when reloading the Directus page after visiting the Nuxtjs page:

![directus-login](https://user-images.githubusercontent.com/105208438/191464388-11354a50-fbd0-4f24-af5b-b61cfd2c35cd.png)

As it is at the moment this would mean, from my understanding, that everyone that visits the webpage can also log into Directus which doesn't make sense to me. How can I authenticate my Nuxtjs app with Directus without giving everyone access and without switching users when visiting the page? Maybe I misunderstood something about the auth mechanism, so any help would be appreciated. Thank you! 

### To Reproduce

1) Clone examples repository (https://github.com/directus/examples)
2) Set up directus and nuxtjs instances
3) Start both services as described in the tutorial / repository
4) Visit http://localhost:8055 and check that you are not logged in
5) Visit http://localhost:3000
6) Reload http://localhost:8055, you are now logged in

### Errors Shown

No errors, see "To Reproduce"

### What version of Directus are you using?

9.16.1

### What version of Node.js are you using?

16.17.0

### What database are you using?

Postgres 13 or SQLite

### What browser are you using?

Firefox, Chrome

### How are you deploying Directus?

locally

**Comment by eikaramba** [comment]
this is absolutely normal. i am not sure how you authenticate, but let's say you use cookies. because your frontend and directus app are running on the same domain the same cookies apply and if you are logged in in one of them you are logged in in both with the same user.

however that does not necessarly mean that a user has ACCESS to the directus app. you can control this in directus and give app access or not for the users role. e.g. if no app access is allowed, if you try to access the actual app it will not work.

i find this actually a cool feature because a logged in user will anyway only see the information in the directus app he is allowed to see. it's just a shortcut to using the API to retrieve informations. but again, you can disallow access to the directus app anyway.

**Reply by julianftw** [reply]
Thank you @eikaramba for your answer! 

To elaborate further on my issue, I already created a limited role "API" account for Nuxtjs and disabled app access which was working fine. However, when my client logs into his admin account and previews the page, he gets switched into the limited account and can't edit content anymore. Following your explanation, this makes sense from a technical point of view. 

However, if my goal would be to have both backend and frontend on the same server and domain, is there any way to both have a limited account for Nuxtjs as well as normal login functionality?

I just tested a workaround which is opening the page in an incognito tab while having Directus open in a normal tab. This disables the login switching.

**Reply by eikaramba** [reply]
unforatuntely it is hard for me to follow you, i am not sure what you mean with "account for nuxtjs". there is just the authenticated user in the browser via cookies/localstorage/etc. and that user token is binded to a role which has app access or not.

the whole API (rest or graphql) is controled via the role access to the collection and has nothing to do with the app access checkbox in directus.

i actually have a sveltekit frontend app and directus backend on the same domain. and in sveltekit i coded a normal login functionality which is setting the cookies for saving the token. you can use directus login/auth methods to handle authentication yourself in the frontend.

i have no experience with nuxtjs resp. i have but not in combination with directus.
